discovery-infra/download_logs.py

```diff
@@ -63,6 +66,13 @@ def download_logs(client: InventoryClient, cluster: dict, dest: str):
     with suppress(assisted_service_client.rest.ApiException):
         client.download_cluster_logs(cluster['id'], os.path.join(output_folder, f"cluster_{cluster['id']}_logs.tar"))
 
+    kubeconfig_path = os.path.join(output_folder, "kubeconfig")
+
+    with suppress(assisted_service_client.rest.ApiException):
+        if cluster['status'] in [ClusterStatus.INSTALLED]:
+            client.download_kubeconfig(cluster['id'], kubeconfig_path)
+            download_must_gather(kubeconfig_path, os.path.join(output_folder, "must-gather"))
```

Comment:
and in finalizing? or if we failed cause of finalizing timeout?

Reply:
Can we gather must-gather on finalizing?

Reply:
yes